src/main/java/spoon/support/compiler/jdt/JDTCommentBuilder.java

```diff
@@ -375,10 +375,16 @@ public void visitCtIf(CtIf e) {
 				}
 				CtStatement elseStatement = e.getElseStatement();
 				if (elseStatement != null && thenStatement != null) {
-					CtStatement thenExpression = ((CtBlock) thenStatement).getStatement(0);
-					CtStatement elseExpression = ((CtBlock) thenStatement).getStatement(0);
-					SourcePosition thenPosition = thenStatement.getPosition().isValidPosition() ? thenStatement.getPosition() : thenExpression.getPosition();
-					SourcePosition elsePosition = elseStatement.getPosition().isValidPosition() ? elseStatement.getPosition() : elseExpression.getPosition();
+					SourcePosition thenPosition = thenStatement.getPosition();
+					if (!thenPosition.isValidPosition()) {
+						CtStatement thenExpression = ((CtBlock) thenStatement).getStatement(0);
```

Comment:
thenStatement isn't always CtBlock in this context.

Reply:
Ok, I see, but it's exactly the way the original code was written (I did not really want to change it):
```
SourcePosition thenPosition = e.getThenStatement().getPosition().isValidPosition() == false ? ((CtBlock) e.getThenStatement()).getStatement(0).getPosition() : e.getThenStatement().getPosition();
SourcePosition elsePosition = e.getElseStatement().getPosition().isValidPosition() == false ? ((CtBlock) e.getElseStatement()).getStatement(0).getPosition() : e.getElseStatement().getPosition();
```
Would you like me to add an additional check like `thenStatement instanceof CtBlock`?